firebase-installations/src/androidTest/java/com/google/firebase/installations/FirebaseInstallationsInstrumentedTest.java

```diff
@@ -266,6 +268,73 @@ public void testGetId_PersistedFidOk_BackendError() throws Exception {
         .isEqualTo(PersistedFid.RegistrationStatus.REGISTER_ERROR);
   }
 
+  @Test
+  public void testGetId_fidRegistrationUncheckedException_statusUpdated() throws Exception {
+    // Mocking unchecked exception on FIS createFirebaseInstallation
+    when(mockClient.createFirebaseInstallation(anyString(), anyString(), anyString(), anyString()))
+        .thenAnswer(
+            invocation -> {
+              throw new InterruptedException();
+            });
+
+    FirebaseInstallations firebaseInstallations =
+        new FirebaseInstallations(
+            mockClock, executor, firebaseApp, mockClient, persistedFid, mockUtils);
+
+    Tasks.await(firebaseInstallations.getId());
+
+    PersistedFidEntry entryValue = persistedFid.readPersistedFidEntryValue();
+    assertWithMessage("Persisted Fid doesn't match")
+        .that(entryValue.getFirebaseInstallationId())
+        .isEqualTo(TEST_FID_1);
+    assertTrue(" Fid Registration Status doesn't match", entryValue.isPending());
+
+    // Waiting for Task that registers FID on the FIS Servers
+    executor.awaitTermination(1000, TimeUnit.MILLISECONDS);
+
+    // Validate that registration status is REGISTER_ERROR
+    PersistedFidEntry updatedFidEntry = persistedFid.readPersistedFidEntryValue();
+    assertWithMessage("Persisted Fid doesn't match")
+        .that(updatedFidEntry.getFirebaseInstallationId())
+        .isEqualTo(TEST_FID_1);
+    assertTrue(" Fid Registration Status doesn't match", updatedFidEntry.isErrored());
```

Comment:
given the "Fid Registration Status doesn't match" message is repeated and could be clarified / improved, how about making an assertion helper? `assertIsErrored` or similar that logs both the actual status and what it was expected to be?

Reply:
Updated. PTAL. Used the PersistedFidEntrySubject.